### ISSUE TITLE
Documentation is very hard to understand for a first time user.

### ISSUE THREAD

**Issue by Tandolf** [terraform-google-modules/terraform-example-foundation#969]
### TL;DR

I would like a bit more documentation about the purpose of certain actions.

### Details

As a first time user of this terraform project, I find the documentation quite hard to understand. 

To get the bootstrap project up and running there is a 17 bullet point list that you need to complete. And many of the actions described are not always documented/clear as to why it should be done.

Some examples:

- in the tfvars its optional to list groups you want created. But in the `groups` setting there is a `billing_project` setting. What is this? why is it placed among the groups? i know what a `billing account` is, but what is a `billing_project`? and why is it needed to create groups? the docs are very unclear here.

- after i have run the bootstrap project, you are asked to clone some policies from a gcp source repo and add these to your terraform project. There is no reference to why i need to do this or at least i can't find any documentation about it.

- you are asked to push your code `Save 0-bootstrap Terraform configuration to gcp-bootstrap source repository:` also no explanation why i need to save my source there. Do i need to save it there? i was thinking of hosting my code on github. Is that possible, or do i have to push my code to gcp source repository? or is this just an opinionated setup?

- there is some documentation asking me to output some variables because i need then in module 3 and 4, which is sort of also strange. I havn't even gotten through module 0 and have a hard time understanding all at the first step and you are asking for things in module 3 and 4, whats the purpose? can i do this at a later time?

As i first time user of this repo i just wanted to tell about my difficulties.

So what im sort of asking for is:
- a bit more purpose for each step, why does this need to be done?
- The documentation on the first page gives a short overview of what does module does. The readme for each module should include more in depth information of what the module _actually_ does.
- What steps are mandatory, can i use github, gitlab? or do need to use gcps source repo? if so why?
- what are some of the parameters, and better explanations of their purpose, why is there a dockerfile in the repo? why is there a sh script in the repo? a bit more about the implementation details.
- first time i ran this gcloud wanted me to install beta, why is that needed?

Thank you

**Comment by andytheapedemontague** [comment]
One thing I'm specifically interested in is the question about if it's possible to use GitHub/GitLab to host the repos or if CSR is required. It appears that when using the Jenkins setup, one can use a private Git server, but it's not clear to me how to do that and still use the Cloud Build module. I'd have a preference for using GitHub since that's where we store all our other code and we can leverage PRs etc but I'd want to stay with Cloud Build. I've taken a look through the code and the CSR dependency does seem quite baked in.
